app/hotels/src/allHotels/searchForm/SearchForm.js

```diff
@@ -50,16 +52,57 @@ class SearchForm extends React.Component<Props> {
   handleCheckoutChange = (date: string) =>
     this.handleDateChange(date, 'checkout');
 
-  handleDateChange = (datestring: string, type: 'checkin' | 'checkout') => {
-    const date = moment(datestring, DISPLAY_DATE_FORMAT).toDate();
+  handleDateChange = (datestring: string, type: DateInputType) => {
+    const date = moment(datestring, DISPLAY_DATE_FORMAT);
+    const checkin =
+      type === 'checkin' ? date : moment(this.props.search.checkin);
+    const checkout =
+      type === 'checkout' ? date : moment(this.props.search.checkout);
+    const whatToAdjust = type === 'checkin' ? 'checkout' : 'checkin';
+    const dates = this.getValidDates(checkin, checkout, whatToAdjust);
 
-    this.props.onChange({ [type]: date });
+    if (dates) {
+      this.props.onChange(dates);
+    }
   };
 
   handleGuestsChange = (roomsConfiguration: RoomConfigurationType) => {
     this.props.onChange({ roomsConfiguration });
   };
 
+  getValidDates = (
+    checkin: moment,
+    checkout: moment,
+    whatToAdjust: DateInputType,
+  ) => {
+    if (checkin === null || checkout === null) {
+      return null;
+    }
+    const diff = moment(checkout).diff(checkin, 'days');
+    if (diff < 1 || diff > 10) {
```

Comment:
I don't understand this condition. Can you make it more clear?

Reply:
fixed